src/FormattingSettingsService.ts

```diff
@@ -1,26 +1,32 @@
 import powerbi from "powerbi-visuals-api";
-import * as formattingSettings from "./FormattingSettingsComponents";
+import { Card, Model, Slice, ToggleSwitch } from "./FormattingSettingsComponents";
+import { IFormattingSettingsCard, IFormattingSettingsService } from "./FormattingSettingsInterfaces";
 
 import visuals = powerbi.visuals;
+import ILocalizationManager = powerbi.extensibility.ILocalizationManager;
 
+export class FormattingSettingsService implements IFormattingSettingsService {
+    private localizationManager: ILocalizationManager;
 
-export class FormattingSettingsModel {
-    cards: Array<formattingSettings.Card>;
+    constructor(localizationManager?: ILocalizationManager) {
```

Comment:
why not have localizationManager be mandatory?

Reply:
Because not all custom visuals use localization
Some of them dont add string resources json (the json file with all translation key and thier value), and they only add displayName without displayNameKey which means they dont use localization